hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/dsl/operations/transactions/AuthorizeContractOperation.java

```diff
@@ -87,9 +88,24 @@ protected SpecOperation computeDelegate(@NonNull final HapiSpec spec) {
             case SpecAccount account -> cryptoUpdate(account.name()).key(managedKeyName);
             case SpecToken token -> {
                 final var op = tokenUpdate(token.name()).adminKey(managedKeyName);
+                if (extraTokenAuthorizations.contains(TokenKeyType.KYC_KEY)) {
```

Comment:
Extended this as it was only including the supply keys.
@tinker-michaelj @Neeharika-Sompalli Can you provide some feedback if it is ok to do it this way?

Reply:
Yes makes a lot of sense @stoyanov-st !